libraries/AP_GPS/AP_GPS_UAVCAN.cpp

```diff
@@ -103,6 +103,10 @@ void AP_GPS_UAVCAN::subscribe_msgs(AP_UAVCAN* ap_uavcan)
 
     uavcan::Subscriber<uavcan::equipment::gnss::Fix, FixCb> *gnss_fix;
     gnss_fix = new uavcan::Subscriber<uavcan::equipment::gnss::Fix, FixCb>(*node);
+    if (gnss_fix == nullptr) {
+        AP_HAL::panic("AP_GPS_UAVCAN: Unable to allocate memory for gnss_fix subscriber");
```

Comment:
a wrapper around config_error() called allocation_error() that just takes a string for the item being allocated would save flash, and allow users to disable the failing subsystem